pkg/kubelet/dockertools/manager.go

```diff
@@ -1336,7 +1336,7 @@ func (dm *DockerManager) KillPod(pod *api.Pod, runningPod kubecontainer.Pod) err
 			glog.Errorf("Failed tearing down the infra container: %v", err)
 			errs <- err
 		}
-		if err := dm.KillContainerInPod(networkContainer.ID, networkSpec, pod); err != nil {
+		if err := dm.KillContainerInPod(networkContainer.ID, networkSpec, pod, "Need to kill pod."); err != nil {
```

Comment:
This is for distinguishing killing container for restarting container and terminating whole Pod. From this point, it's not generic. We can get more details in log file but we can't get Pod delete  event now. I considered logging specific killPod reason also. But I found it had several kinds of reason and some killPod function call was caused by containerRuntime internal logic which may different between docker and rkt. So I logging a generic killPod reason for now. Surely logging specific killPod reason is better. I think we can record Pod create/delete event to improve it in future. What do you think?